Modules/Core/Common/include/itkMacro.h

```diff
@@ -1271,6 +1271,14 @@ compilers.
   ITK_MACROEND_NOOP_STATEMENT
 
 
+// A useful macro in the PrintSelf method for printing member variables
+// that require to be casted to the corresponding numeric value so that
+// their numeric value gets printed.
+#define itkPrintSelfNumericTraitsMacro(name, type)                                                                    \
+  os << indent << #name << ": " << static_cast<typename NumericTraits<type>::PrintType>(this->m_##name) << std::endl; \
+  ITK_MACROEND_NOOP_STATEMENT
```

Comment:
Thanks for this idea, Jon, but I would prefer to only make _limited_ use of macro's. In general, I think macro's make code harder to understand, and harder to debug. Specifically in this case I would at least suggest to take `os` and `indent` out of the macro definition, because it is unclear from the outside that a call to this macro "captures" these two variables.

Last December I already suggested a different approach for the implementation of those PrintSelf member functions (but I did not have time to make a PR in that direction: https://github.com/InsightSoftwareConsortium/ITK/pull/3802#discussion_r1053351196

Reply:
Niels, it is still WIP, but here my few thoughts:
- After #3872, I think it becomes clear that having a macro saves typing/time and allows to print all members with a consistent style: I'd dare to say that most of the times what this macro attempts to replace has not been done because it requires a non-negligible amount of typing for what can be considered to be a low reward effort at first sight, but it pays off when debugging issues, as @blowekamp made it clear in https://github.com/InsightSoftwareConsortium/ITK/pull/3872#issuecomment-138553199.
- The macro follows the `itkPrintSelfObjectMacro` philosophy, and in the same way, the `os` and `indent` are not taken as parameters as they do not change. But that is the least of the issues for now.
 - The macro is very easy to understand in this case, as it is in the `itkPrintSelfObjectMacro` case.